src/charm.py

```diff
@@ -31,6 +28,10 @@
 logger = logging.getLogger(__name__)
 
 CONFIG_PATH = "/etc/agent/agent.yaml"
+METRICS_RULES_SRC_PATH = "./src/prometheus_alert_rules"
+METRICS_RULES_PATH = "./prometheus_alert_rules"
+REMOTE_WRITE_RELATION_NAME = "send-remote-write"
+SCRAPE_RELATION_NAME = "metrics-endpoint"
```

Comment:
We actually could import https://github.com/canonical/prometheus-k8s-operator/blob/e964e639274efff359de78b65c261c9a9cf981be/lib/charms/prometheus_k8s/v0/prometheus_scrape.py#L362

Reply:
This way feels slightly safer and also easier to read without having to open up prometheus_scrape code.

Reply:
I think what @mmanciop suggests is already practiced elsewhere: the lib is the one responsible for the relation, so if we have to hardcode that literal somewhere (in addition to metadata.yaml), it is probably better to hardcode it only once.

Reply:
This necessarily adds coupling between an external source and code controlled by this charm, though, and `fetch-lib` could cause an unintended divergence between what's in `metadata.yaml` and what's in the code.

Yes, tests would probably catch it, but "our relation name changed" should probably be something which is intentional rather than a side effect.